--- a/Makefile.am	2018-04-16 09:15:05.172748991 +0100
+++ b/Makefile.am	2018-04-16 09:15:43.009935778 +0100
@@ -1,7 +1,7 @@
 SUBDIRS = doc
 
-source_h = xsettings-client.h  xsettings-common.h
-source_c = xsettings-client.c  xsettings-common.c
+source_h = xsettings-client.h
+source_c = xsettings-client.c
 
 INCLUDES = -I $(includedir) $(X_CFLAGS)
 
--- a/Makefile.in	2018-04-16 09:15:05.172748991 +0100
+++ b/Makefile.in	2018-04-16 09:16:02.733032247 +0100
@@ -141,8 +141,8 @@
 target_alias = @target_alias@
 SUBDIRS = doc
 
-source_h = xsettings-client.h  xsettings-common.h
-source_c = xsettings-client.c  xsettings-common.c
+source_h = xsettings-client.h
+source_c = xsettings-client.c
 
 INCLUDES = -I $(includedir) $(X_CFLAGS)
 
