diff --git a/arch/arm64/boot/dts/mediatek/mt8195-cherry-tomato-r1.dts b/arch/arm64/boot/dts/mediatek/mt8195-cherry-tomato-r1.dts
index 11fc83ddf..adcb3deba 100644
--- a/arch/arm64/boot/dts/mediatek/mt8195-cherry-tomato-r1.dts
+++ b/arch/arm64/boot/dts/mediatek/mt8195-cherry-tomato-r1.dts
@@ -27,6 +27,10 @@ &pcie0 {
 	pinctrl-0 = <&pcie0_pins_default>;
 };
 
+&pciephy {
+	status = "okay";
+};
+
 &ts_10 {
 	status = "okay";
 };
