From f35dc4fceeecf3051ba46a19356fa8d67793da50 Mon Sep 17 00:00:00 2001
From: Martin Klapetek <mklapetek@kde.org>
Date: Wed, 24 Feb 2016 15:45:03 -0500
Subject: [PATCH] [KTp] Make SpaceBar the preferred text channel handler

---
 KTp/actions.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/KTp/actions.cpp b/KTp/actions.cpp
index 17d631f..8a29171 100644
--- a/KTp/actions.cpp
+++ b/KTp/actions.cpp
@@ -32,7 +32,7 @@
 #include <KNotification>
 #include <KAboutData>
 
-#define PREFERRED_TEXT_CHAT_HANDLER QLatin1String("org.freedesktop.Telepathy.Client.KTp.TextUi")
+#define PREFERRED_TEXT_CHAT_HANDLER QLatin1String("org.freedesktop.Telepathy.Client.SpaceBar")
 #define PREFERRED_FILE_TRANSFER_HANDLER QLatin1String("org.freedesktop.Telepathy.Client.KTp.FileTransfer")
 #define PREFERRED_AUDIO_VIDEO_HANDLER QLatin1String("org.freedesktop.Telepathy.Client.KTp.CallUi")
 #define PREFERRED_RFB_HANDLER QLatin1String("org.freedesktop.Telepathy.Client.krfb_rfb_handler")
-- 
2.25.0

