From 5697e58578f1af224d13fc40638a210f2da659d3 Mon Sep 17 00:00:00 2001
From: root <root@computer.craftyguy.net>
Date: Wed, 9 Jan 2019 00:02:17 +0000
Subject: [PATCH] remove dependency on git

---
 Makefile | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 4b449ad..36a9ffd 100644
--- a/Makefile
+++ b/Makefile
@@ -33,8 +33,7 @@ $(MKIMG): buildinfo $(SRCS)
 bin: $(MKIMG)
 
 buildinfo:
-	@echo -n '#define MKIMAGE_COMMIT 0x' > src/build_info.h
-	@git rev-parse --short=8 HEAD >> src/build_info.h
+	@echo -n '#define MKIMAGE_COMMIT 0x7bb91090' > src/build_info.h
 	@echo '' >> src/build_info.h
 
 help:
-- 
2.20.1

