diff --git a/arch/arm/boot/dts/sprd-scx35l_sharkls_j3xnlte_rev00.dts b/arch/arm/boot/dts/sprd-scx35l_sharkls_j3xnlte_rev00.dts
index 6d277064..ffad458c 100755
--- a/arch/arm/boot/dts/sprd-scx35l_sharkls_j3xnlte_rev00.dts
+++ b/arch/arm/boot/dts/sprd-scx35l_sharkls_j3xnlte_rev00.dts
@@ -34,7 +34,7 @@
 / {
 	model = "Spreadtrum SP8835EB board";
 	compatible = "sprd,sp8835eb";
-	sprd,sc-id = <9830 0 0x20000>;
+	qcom,msm-id = <9830 0 0x20000>;
 	#address-cells = <1>;
 	#size-cells = <1>;
 	interrupt-parent = <&gic>;
diff --git a/arch/arm/boot/dts/sprd-scx35l_sharkls_j3xnlte_rev01.dts b/arch/arm/boot/dts/sprd-scx35l_sharkls_j3xnlte_rev01.dts
index 45b86b00..4f0e95ba 100755
--- a/arch/arm/boot/dts/sprd-scx35l_sharkls_j3xnlte_rev01.dts
+++ b/arch/arm/boot/dts/sprd-scx35l_sharkls_j3xnlte_rev01.dts
@@ -34,7 +34,7 @@
 / {
 	model = "Spreadtrum SP8835EB board";
 	compatible = "sprd,sp8835eb";
-	sprd,sc-id = <9830 1 0x20000>;
+	qcom,msm-id = <9830 1 0x20000>;
 	#address-cells = <1>;
 	#size-cells = <1>;
 	interrupt-parent = <&gic>;
diff --git a/arch/arm/boot/dts/sprd-scx35l_sharkls_j3xnlte_rev02.dts b/arch/arm/boot/dts/sprd-scx35l_sharkls_j3xnlte_rev02.dts
index cd10c56c..e2d50b43 100755
--- a/arch/arm/boot/dts/sprd-scx35l_sharkls_j3xnlte_rev02.dts
+++ b/arch/arm/boot/dts/sprd-scx35l_sharkls_j3xnlte_rev02.dts
@@ -34,7 +34,7 @@
 / {
 	model = "Spreadtrum SP8835EB board";
 	compatible = "sprd,sp8835eb";
-	sprd,sc-id = <9830 2 0x20000>;
+	qcom,msm-id = <9830 2 0x20000>;
 	#address-cells = <1>;
 	#size-cells = <1>;
 	interrupt-parent = <&gic>;
diff --git a/arch/arm/boot/dts/sprd-scx35l_sharkls_j3xnlte_rev03.dts b/arch/arm/boot/dts/sprd-scx35l_sharkls_j3xnlte_rev03.dts
index b38713ce..3bf64b03 100755
--- a/arch/arm/boot/dts/sprd-scx35l_sharkls_j3xnlte_rev03.dts
+++ b/arch/arm/boot/dts/sprd-scx35l_sharkls_j3xnlte_rev03.dts
@@ -34,7 +34,7 @@
 / {
 	model = "Spreadtrum SP8835EB board";
 	compatible = "sprd,sp8835eb";
-	sprd,sc-id = <9830 3 0x20000>;
+	qcom,msm-id = <9830 3 0x20000>;
 	#address-cells = <1>;
 	#size-cells = <1>;
 	interrupt-parent = <&gic>;
diff --git a/arch/arm/boot/dts/sprd-scx35l_sharkls_j3xnlte_rev04.dts b/arch/arm/boot/dts/sprd-scx35l_sharkls_j3xnlte_rev04.dts
index 393d2db1..c339ceae 100755
--- a/arch/arm/boot/dts/sprd-scx35l_sharkls_j3xnlte_rev04.dts
+++ b/arch/arm/boot/dts/sprd-scx35l_sharkls_j3xnlte_rev04.dts
@@ -34,7 +34,7 @@
 / {
 	model = "Spreadtrum SP8835EB board";
 	compatible = "sprd,sp8835eb";
-	sprd,sc-id = <9830 4 0x20000>;
+	qcom,msm-id = <9830 4 0x20000>;
 	#address-cells = <1>;
 	#size-cells = <1>;
 	interrupt-parent = <&gic>;
diff --git a/arch/arm/boot/dts/sprd-scx35l_sharkls_j3xnlte_rev05.dts b/arch/arm/boot/dts/sprd-scx35l_sharkls_j3xnlte_rev05.dts
index 5611b8bb..bb1d2d3a 100755
--- a/arch/arm/boot/dts/sprd-scx35l_sharkls_j3xnlte_rev05.dts
+++ b/arch/arm/boot/dts/sprd-scx35l_sharkls_j3xnlte_rev05.dts
@@ -34,7 +34,7 @@
 / {
 	model = "Spreadtrum SP8835EB board";
 	compatible = "sprd,sp8835eb";
-	sprd,sc-id = <9830 5 0x20000>;
+	qcom,msm-id = <9830 5 0x20000>;
 	#address-cells = <1>;
 	#size-cells = <1>;
 	interrupt-parent = <&gic>;
