From e11f599d4985fdf516733423cf086e6454902e50 Mon Sep 17 00:00:00 2001
From: Robill Tian Supatrio <rsupatrio@android-x86.org>
Date: Sun, 8 Jan 2012 12:51:32 +0800
Subject: [PATCH] Fixed compilation error for rndis_bind_config.

---
 drivers/usb/gadget/ether.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/usb/gadget/ether.c b/drivers/usb/gadget/ether.c
index 1690c9d..53ee2c5 100644
--- a/drivers/usb/gadget/ether.c
+++ b/drivers/usb/gadget/ether.c
@@ -246,7 +246,7 @@ static int __init rndis_do_config(struct usb_configuration *c)
 		c->bmAttributes |= USB_CONFIG_ATT_WAKEUP;
 	}
 
-	return rndis_bind_config(c, hostaddr);
+	return rndis_bind_config(c, hostaddr, 12345, "android-arm");
 }
 
 static struct usb_configuration rndis_config_driver = {
-- 
1.7.4.1

