diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index ae27484..e488487 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,4 +1,10 @@
+
+set(QTWEBENGINE_REAL_PRIVATE_INCLUDES "/usr/include/QtWebEngine/${Qt5WebEngine_VERSION}")
+set(QTWEBENGINE_REAL_PRIVATE_INCLUDES2 "/usr/include/QtWebEngine/${Qt5WebEngine_VERSION}/QtWebEngine")
+MESSAGE( STATUS "CMAKE_BINARY_DIR:         " ${QTWEBENGINE_REAL_PRIVATE_INCLUDES} )
 include_directories(
+    ${QTWEBENGINE_REAL_PRIVATE_INCLUDES}
+    ${QTWEBENGINE_REAL_PRIVATE_INCLUDES2}
     ${Qt5WebEngine_PRIVATE_INCLUDE_DIRS}
     ${Qt5WebChannel_PRIVATE_INCLUDE_DIRS}
     ${Qt5Quick_PRIVATE_INCLUDE_DIRS}
