diff --git a/src/qml/ApplicationContainer.qml b/src/qml/ApplicationContainer.qml
index 3d5f332..f90f5d4 100644
--- a/src/qml/ApplicationContainer.qml
+++ b/src/qml/ApplicationContainer.qml
@@ -127,7 +127,6 @@ Flickable {
             backgroundColor: (webAppWindow.windowType === "dashboard" || webAppWindow.windowType === "popupalert") ? "transparent": "white"
 
             userScripts: webAppWindow.userScripts;
-            devicePixelRatio: webAppWindow.devicePixelRatio
 
             onJavaScriptConsoleMessage: console.warn("CONSOLE JS: " + message);
 
