diff --git a/usr/u8500_initramfs.list b/usr/u8500_initramfs.list
index 35e8d68..c846052 100644
--- a/usr/u8500_initramfs.list
+++ b/usr/u8500_initramfs.list
@@ -29,5 +29,3 @@ file /stage1/init source/usr/u8500_initramfs_files/init 755 0 0
 dir /sys 755 0 0
 dir /mnt 755 0 0
 dir /mnt/.lfs 755 0 0
-file /stage1/boot.cpio ../../ramdisk.cpio 644 0 0
-file /stage1/recovery.cpio ../../ramdisk-recovery.cpio 644 0 0
