diff --git a/GSettings/gsettings-qt.pro b/GSettings/gsettings-qt.pro
index 218eeea..6f1cd3e 100644
--- a/GSettings/gsettings-qt.pro
+++ b/GSettings/gsettings-qt.pro
@@ -25,6 +25,10 @@ INSTALLS += extra
 
 qmltypes.path = $$installPath
 qmltypes.files = plugins.qmltypes
-qmltypes.extra = export LD_PRELOAD=../src/libgsettings-qt.so.1; $$[QT_INSTALL_BINS]/qmlplugindump -notrelocatable GSettings 1.0 .. > $(INSTALL_ROOT)/$$installPath/plugins.qmltypes
+# TODO: Complain at whoever is responsible for this - I lost an hour debugging this
+# Relative and absolute LD_PRELOAD path doesn't work
+# Relative LD_LIBRARY_PATH doesn't work
+# Absolute LD_LIBRARY_PATH DOES work
+qmltypes.extra = export LD_LIBRARY_PATH=$$PWD/../src; $$[QT_INSTALL_BINS]/qmlplugindump -notrelocatable GSettings 1.0 .. > $(INSTALL_ROOT)/$$installPath/plugins.qmltypes
 INSTALLS += qmltypes
 
