diff -Naur a/libmb.pc.in b/libmb.pc.in
--- a/libmb.pc.in	2016-07-08 11:15:21.000000000 +0100
+++ b/libmb.pc.in	2018-05-08 04:24:57.673377262 +0100
@@ -7,6 +7,8 @@
 Description: Utility Library used by Matchbox utilities.
 Version: @VERSION@
 
-Requires: @XLIBS_REQUIRED@ @PANGO_REQUIRED@ @PNG_REQUIRED@ @XFT_REQUIRED@
-Libs: -L${libdir} -lmb @MB_EXTRA_LIBS@
+Requires.private: @XLIBS_REQUIRED@ @PANGO_REQUIRED@ @PNG_REQUIRED@
+Requires: libxsettings-client
+Libs: -L${libdir} -lmb
+Libs.private: @MB_EXTRA_LIBS@
 Cflags: -I${includedir} @MB_EXTRA_CFLAGS@
