Upstream pull request: https://github.com/andersson/rmtfs/pull/17

diff --git a/storage.c b/storage.c
index 107b296..2c5f2f5 100644
--- a/storage.c
+++ b/storage.c
@@ -41,6 +41,7 @@ static const struct partition partition_table[] = {
 	{ "/boot/modem_fs2", "modem_fs2", "modemst2" },
 	{ "/boot/modem_fsc", "modem_fsc", "fsc" },
 	{ "/boot/modem_fsg", "modem_fsg", "fsg" },
+	{ "/boot/modem_tng", "modem_tng", "tunning" },
 	{ "/boot/modem_tunning", "modem_tunning", "tunning" },
 	{}
 };
