diff --git a/arch/arm/mach-msm/restart.c b/arch/arm/mach-msm/restart.c
index ab211fb0009..61ac4160949 100755
--- a/arch/arm/mach-msm/restart.c
+++ b/arch/arm/mach-msm/restart.c
@@ -452,7 +452,9 @@ void msm_restart(char mode, const char *cmd)
         mdelay(1000);
     }*/
 #endif /* VENDOR_EDIT*/
-
+	pr_info("%s : flush console and then delay 1s waiting log printing", __func__);
+	arm_machine_flush_console();
+	mdelay(1000);
 	if (!use_restart_v2()) {
 		__raw_writel(0, msm_tmr0_base + WDT0_EN);
 		if (!(machine_is_msm8x60_fusion() ||
diff --git a/drivers/input/keyboard/gpio_keys.c b/drivers/input/keyboard/gpio_keys.c
index 2e1f3adbbe8..9a81afad423 100644
--- a/drivers/input/keyboard/gpio_keys.c
+++ b/drivers/input/keyboard/gpio_keys.c
@@ -30,10 +30,10 @@
 #include <linux/of_gpio.h>
 #include <linux/spinlock.h>
 
-#ifdef VENDOR_EDIT
+//#ifdef VENDOR_EDIT
 //hefaxi@filesystems, 2015/07/03, add for force dump function
-#include <linux/oem_force_dump.h>
-#endif
+//#include <linux/oem_force_dump.h>
+//#endif
 
 struct gpio_button_data {
 	const struct gpio_keys_button *button;
@@ -338,7 +338,7 @@ static void gpio_keys_gpio_report_event(struct gpio_button_data *bdata)
 
 #ifdef VENDOR_EDIT
 //hefaxi@filesystems, 2015/07/03, add for force dump function
-    oem_check_force_dump_key(button->code,state);
+    //oem_check_force_dump_key(button->code,state);
 #endif
 
 	if (type == EV_ABS) {
diff --git a/drivers/platform/msm/qpnp-power-on.c b/drivers/platform/msm/qpnp-power-on.c
index c6aae05a514..1de7f84fd01 100755
--- a/drivers/platform/msm/qpnp-power-on.c
+++ b/drivers/platform/msm/qpnp-power-on.c
@@ -24,10 +24,10 @@
 #include <linux/log2.h>
 #include <linux/qpnp/power-on.h>
 
-#ifdef VENDOR_EDIT
+//#ifdef VENDOR_EDIT
 //hefaxi@filesystems, 2015/07/03, add for force dump function
-#include <linux/oem_force_dump.h>
-#endif
+//#include <linux/oem_force_dump.h>
+//#endif
 
 #define PMIC_VER_8941           0x01
 #define PMIC_VERSION_REG        0x0105
@@ -438,10 +438,10 @@ qpnp_pon_input_dispatch(struct qpnp_pon *pon, u32 pon_type)
 					(pon_rt_sts & pon_rt_bit));
 	input_sync(pon->pon_input);
 
-#ifdef VENDOR_EDIT
+//#ifdef VENDOR_EDIT
 //hefaxi@filesystems, 2015/07/03, add for force dump function
-    oem_check_force_dump_key(cfg->key_code,(pon_rt_sts & pon_rt_bit));
-#endif
+    //oem_check_force_dump_key(cfg->key_code,(pon_rt_sts & pon_rt_bit));
+//#endif
 	return 0;
 }
 
