From f1bbc6993220921b30e59bf5091952c6400bf219 Mon Sep 17 00:00:00 2001
From: l10n daemon script <scripty@kde.org>
Date: Sat, 7 Apr 2018 07:47:38 +0000
Subject: [PATCH] GIT_SILENT Upgrade ECM and KF5 version requirements for
 5.45.0 release.

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b018352..a8d8314 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,7 +5,7 @@ project(KI18n VERSION ${KF5_VERSION})
 
 # ECM setup
 include(FeatureSummary)
-find_package(ECM 5.44.0  NO_MODULE)
+find_package(ECM 5.45.0  NO_MODULE)
 set_package_properties(ECM PROPERTIES TYPE REQUIRED DESCRIPTION "Extra CMake Modules." URL "https://projects.kde.org/projects/kdesupport/extra-cmake-modules")
 feature_summary(WHAT REQUIRED_PACKAGES_NOT_FOUND FATAL_ON_MISSING_REQUIRED_PACKAGES)
 
From 94acc129866640ab430dd462ea5c2678939d80d5 Mon Sep 17 00:00:00 2001
From: l10n daemon script <scripty@kde.org>
Date: Sat, 10 Mar 2018 08:40:07 +0000
Subject: [PATCH] GIT_SILENT Upgrade KF5 version to 5.45.0.

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f345ac2..4b4df4a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 3.0)
 
-set(KF5_VERSION "5.44.0") # handled by release scripts
+set(KF5_VERSION "5.45.0") # handled by release scripts
 project(KI18n VERSION ${KF5_VERSION})
 
 # ECM setup

