diff --git a/configure.ac b/configure.ac
index 31134c3..412bce9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -6,7 +6,7 @@ AM_INIT_AUTOMAKE([foreign -Wall silent-rules])
 AC_CONFIG_MACRO_DIR(m4)
 AM_SILENT_RULES([yes])
 
-AS_IF([! test -n "$VERSION"], [
+AS_IF([! test -n "$PACKAGE_VERSION"], [
    AC_MSG_ERROR([git-version-gen failed])
 ])
 
@@ -20,9 +20,9 @@ AC_PROG_CC_C99
 AM_PROG_CC_C_O
 AC_PROG_GCC_TRADITIONAL
 
-m4_define(pa_major, `echo $VERSION | cut -d. -f1 | cut -d- -f1`)
-m4_define(pa_minor, `echo $VERSION | cut -d. -f2 | cut -d- -f1`)
-m4_define(pa_module_version, `echo $VERSION | cut -d. -f3 | cut -d- -f1`)
+m4_define(pa_major, `echo $PACKAGE_VERSION | cut -d. -f1 | cut -d- -f1`)
+m4_define(pa_minor, `echo $PACKAGE_VERSION | cut -d. -f2 | cut -d- -f1`)
+m4_define(pa_module_version, `echo $PACKAGE_VERSION | cut -d. -f3 | cut -d- -f1`)
 
 AC_SUBST(PA_MAJOR, pa_major)
 AC_SUBST(PA_MAJORMINOR, pa_major.pa_minor)
@@ -326,7 +326,7 @@ AC_OUTPUT
 
 echo "
 
- ---{ $PACKAGE_NAME $VERSION }---
+ ---{ $PACKAGE_NAME $PACKAGE_VERSION }---
 
     compiler:               ${CC}
     CFLAGS:                 ${CFLAGS}
