diff --git a/plugin/CMakeLists.txt b/plugin/CMakeLists.txt
index 36aef53..bb7b38c 100644
--- a/plugin/CMakeLists.txt
+++ b/plugin/CMakeLists.txt
@@ -1,4 +1,9 @@
+set(QTWEBENGINE_REAL_PRIVATE_INCLUDES "/usr/include/QtWebEngine/${Qt5WebEngine_VERSION}")
+set(QTWEBENGINE_REAL_PRIVATE_INCLUDES2 "/usr/include/QtWebEngine/${Qt5WebEngine_VERSION}/QtWebEngine")
+
 include_directories(
+        ${QTWEBENGINE_REAL_PRIVATE_INCLUDES}
+        ${QTWEBENGINE_REAL_PRIVATE_INCLUDES2}
 	${Qt5WebEngine_PRIVATE_INCLUDE_DIRS}
 	${Qt5Quick_PRIVATE_INCLUDE_DIRS})
 
