From f31164cc9bd71a5ba613b2323a462a829dea83aa Mon Sep 17 00:00:00 2001
From: Yuri Edward <yuri6037@outlook.com>
Date: Sun, 10 Jan 2021 23:54:46 +0100
Subject: [PATCH 3/3] Bumped version of libhandy to 1.0.1

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 3f2772d8..2ce78816 100644
--- a/meson.build
+++ b/meson.build
@@ -88,7 +88,7 @@ gthread = dependency('gthread-2.0', version: '>=' + target_glib)
 iso_codes = dependency('iso-codes')
 javascriptcoregtk = dependency('javascriptcoregtk-4.0', version: '>=' + target_webkit)
 json_glib = dependency('json-glib-1.0', version: '>= 1.0')
-libhandy = dependency('libhandy-1', version: '>= 1.0')
+libhandy = dependency('libhandy-1', version: '>= 1.0.1')
 libmath = cc.find_library('m')
 libpeas = dependency('libpeas-1.0', version: '>= 1.24.0')
 libpeas_gtk = dependency('libpeas-gtk-1.0', version: '>= 1.24.0')
-- 
2.30.2

