From 6a0d7e507709e71361c67d1b923674423bbdd50e Mon Sep 17 00:00:00 2001
From: Volker Krause <vkrause@kde.org>
Date: Wed, 18 Dec 2019 18:35:47 +0100
Subject: [PATCH] Fix build

---
 src/app/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/app/CMakeLists.txt b/src/app/CMakeLists.txt
index ae6b977..8f393cd 100644
--- a/src/app/CMakeLists.txt
+++ b/src/app/CMakeLists.txt
@@ -160,7 +160,7 @@ if (ANDROID)
         weather-storm
     )
 else ()
-    target_link_libraries(itinerary PRIVATE Qt5::Positioning Qt5::DBus)
+    target_link_libraries(itinerary PRIVATE Qt5::Positioning Qt5::DBus Qt5::Widgets)
     target_link_libraries(itinerary-app PRIVATE
         KF5::DBusAddons
         Qt5::Widgets
-- 
2.24.1

