diff --git a/look-and-feel/contents/plasmoidsetupscripts/org.kde.phone.homescreen.js b/look-and-feel/contents/plasmoidsetupscripts/org.kde.phone.homescreen.js
index 5955db9..32d8fcf 100644
--- a/look-and-feel/contents/plasmoidsetupscripts/org.kde.phone.homescreen.js
+++ b/look-and-feel/contents/plasmoidsetupscripts/org.kde.phone.homescreen.js
@@ -1,5 +1,8 @@
 applet.wallpaperPlugin = 'org.kde.image'
 applet.writeConfig("AppOrder", ["org.kde.phone.dialer.desktop", "org.kde.mobile.angelfish.desktop", "org.kde.phone.calindori.desktop", "org.kde.mobile.camera.desktop"])
 applet.writeConfig("Favorites", ["org.kde.phone.dialer.desktop", "org.kde.mobile.angelfish.desktop", "org.kde.phone.calindori.desktop", "org.kde.mobile.camera.desktop"])
+applet.currentConfigGroup = new Array ("Wallpaper", "org.kde.image", "General")
+applet.writeConfig("Image", "file:///usr/share/wallpapers/postmarketos.jpg")
+applet.writeConfig("FillMode", "2")
 applet.reloadConfig()
 
