diff --git a/src/hwcomposer.c b/src/hwcomposer.c
index bb6d204..52e06df 100644
--- a/src/hwcomposer.c
+++ b/src/hwcomposer.c
@@ -123,7 +123,7 @@ Bool hwc_hwcomposer_init(ScrnInfoPtr pScrn)
 		return hwc_hwcomposer2_init(pScrn);
 #endif
 	
-	hwc_composer_device_1_t *hwcDevicePtr = (hwc_composer_device_1_t*) hwcDevice;
+	hwc_composer_device_1_t *hwcDevicePtr = hwc->hwcDevicePtr = (hwc_composer_device_1_t*) hwcDevice;
 	hwc_set_power_mode(pScrn, HWC_DISPLAY_PRIMARY, 1);
 
 	uint32_t configs[5];
