diff --git a/include/location_service/com/ubuntu/location/service/permission_manager.h b/include/location_service/com/ubuntu/location/service/permission_manager.h
index 73b79bd..045a83d 100644
--- a/include/location_service/com/ubuntu/location/service/permission_manager.h
+++ b/include/location_service/com/ubuntu/location/service/permission_manager.h
@@ -19,6 +19,7 @@
 #define LOCATION_SERVICE_COM_UBUNTU_LOCATION_SERVICE_PERMISSION_MANAGER_H_
 
 #include <memory>
+#include <sys/types.h>
 
 namespace com
 {
