diff --git a/bin/kwinwrapper b/bin/kwinwrapper
index 986b9e3..4ec1b18 100755
--- a/bin/kwinwrapper
+++ b/bin/kwinwrapper
@@ -15,6 +15,7 @@ export PLASMA_PLATFORM=phone
 export QT_QUICK_CONTROLS_STYLE=Plasma
 export QT_ENABLE_GLYPH_CACHE_WORKAROUND=1
 export QT_QUICK_CONTROLS_MOBILE=true
+export QT_VIRTUALKEYBOARD_STYLE=Plasma
 
 # work around to fix : https://bugs.launchpad.net/ubuntu/+source/mesa/+bug/1507584
 if [ -d /usr/lib/arm-linux-gnueabihf/libhybris-egl ]; then
