diff --git a/CMakeLists.txt b/CMakeLists.txt
index 609ccac..a0ba2b3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -27,7 +27,6 @@ find_package(Qt5 ${QT_MIN_VERSION} CONFIG REQUIRED Core Gui Widgets Qml Quick Te
 
 find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS Plasma Service Declarative I18n KIO People)
 find_package(KF5 REQUIRED COMPONENTS PlasmaQuick DBusAddons Notifications)
-find_package(TelepathyQt5 REQUIRED)
 find_package(KF5Wayland CONFIG)
 set_package_properties(KF5Wayland PROPERTIES
                        TYPE REQUIRED
@@ -55,6 +54,5 @@ add_subdirectory(bin)
 add_subdirectory(applets)
 add_subdirectory(containments)
  add_subdirectory(components)
-add_subdirectory(dialer)
 add_subdirectory(sounds)
 add_subdirectory(touchscreentest)
