Description: Fix typo and tidy program help text.

--- matchbox-window-manager-1.2-osso21.orig/matchbox-window-manager/src/wm.c
+++ matchbox-window-manager-1.2-osso21/matchbox-window-manager/src/wm.c
@@ -194,7 +194,7 @@ wm_usage(char *progname)
    printf("\t-use_dialog_mode  <free|static|const-horiz>\n");
    printf("\t-use_desktop_mode <decorated|plain>\n");
    printf("\t-use_super_modal  <yes|no>\n");
-   printf("\t-force_dialogs    <comma seperated list of window titles>\n");
+   printf("\t-force_dialogs    <comma separated list of window titles>\n");
 #ifndef NO_KBD
    printf("\t-kbdconfig        <path to option key shortcuts config file>\n");
 #endif
@@ -203,7 +203,7 @@ wm_usage(char *progname)
 #endif
 
 #ifdef STANDALONE
-   printf("\t-titlebar_panel   <x11 geometry>\n");
+   printf("\t-titlebar_panel   <X11 geometry>\n");
 #endif
    printf("\n");
    printf("Compile time features;\n");
@@ -244,15 +244,15 @@ wm_usage(char *progname)
 #endif
 
 #ifdef MB_HAVE_JPEG
-   printf("\tJPG support                      yes\n");
+   printf("\tJPEG support                     yes\n");
 #else
-   printf("\tJPG support                      no\n");
+   printf("\tJPEG support                     no\n");
 #endif
 
 #ifdef MB_HAVE_PANGO
-   printf("\tpango support                    yes\n");
+   printf("\tPango support                    yes\n");
 #else
-   printf("\tpango support                    no\n");
+   printf("\tPango support                    no\n");
 #endif
 
 #ifndef STANDALONE
@@ -268,15 +268,15 @@ wm_usage(char *progname)
 #endif
 
 #ifdef USE_COMPOSITE
-   printf("\tcomposite support                yes\n");
+   printf("\tComposite support                yes\n");
 #else
-   printf("\tcomposite support                no\n");
+   printf("\tComposite support                no\n");
 #endif
 
 #ifndef NO_PING
-   printf("\tping protocol support            yes\n");
+   printf("\tPing protocol support            yes\n");
 #else
-   printf("\tping protocol support            no\n");
+   printf("\tPing protocol support            no\n");
 #endif
 
 #ifndef NO_XRM
