From 195b1850c7071de835c73de668a0437085db48d4 Mon Sep 17 00:00:00 2001
From: Robert Yang <decatf@gmail.com>
Date: Fri, 11 Jan 2019 14:26:35 -0500
Subject: [PATCH] meson: correct gallium grate variable

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 8b796ba79c3..1ad8eafb117 100644
--- a/meson.build
+++ b/meson.build
@@ -1049,7 +1049,7 @@ _libdrm_checks = [
   ['nouveau', (with_gallium_nouveau or with_dri_nouveau)],
   ['etnaviv', with_gallium_etnaviv],
   ['freedreno', with_gallium_freedreno],
-  ['tegra', with_gallium_tegra],
+  ['tegra', with_gallium_grate],
 ]
 
 # Loop over the enables versions and get the highest libdrm requirement for all
