From d533086825e713e3c13ce8b581f4ee86e5ae11d8 Mon Sep 17 00:00:00 2001
From: Ion Agorria <ion@agorria.com>
Date: Sat, 19 Jan 2019 12:17:10 +0100
Subject: [PATCH] Warn only once about skb_release_head_state in irq

---
 net/core/skbuff.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/core/skbuff.c b/net/core/skbuff.c
index 31d2aad99ae4..fafe8d85b7e2 100644
--- a/net/core/skbuff.c
+++ b/net/core/skbuff.c
@@ -613,7 +613,7 @@ static void skb_release_head_state(struct sk_buff *skb)
 	secpath_put(skb->sp);
 #endif
 	if (skb->destructor) {
-		WARN_ON(in_irq());
+		WARN_ON_ONCE(in_irq());
 		skb->destructor(skb);
 	}
 #if IS_ENABLED(CONFIG_NF_CONNTRACK)
-- 
2.19.2

