This patch fixed 'No valid dts found' error.
diff --git a/arch/arm/boot/dts/sprd-scx20_j1mini3g_r00.dts b/arch/arm/boot/dts/sprd-scx20_j1mini3g_r00.dts
index e87c172f..38f95e69 100644
--- a/arch/arm/boot/dts/sprd-scx20_j1mini3g_r00.dts
+++ b/arch/arm/boot/dts/sprd-scx20_j1mini3g_r00.dts
@@ -33,7 +33,7 @@
 / {
 	model = "Spreadtrum SP8835EB board";
 	compatible = "sprd,sp8835eb";
-	sprd,sc-id = <8830 0 0x20000>;
+	qcom,msm-id = <8830 0 0x20000>;
 	#address-cells = <1>;
 	#size-cells = <1>;
 	interrupt-parent = <&gic>;
diff --git a/arch/arm/boot/dts/sprd-scx20_j1mini3g_r01.dts b/arch/arm/boot/dts/sprd-scx20_j1mini3g_r01.dts
index 55a729d0..caff8fe9 100644
--- a/arch/arm/boot/dts/sprd-scx20_j1mini3g_r01.dts
+++ b/arch/arm/boot/dts/sprd-scx20_j1mini3g_r01.dts
@@ -33,7 +33,7 @@
 / {
 	model = "Spreadtrum SP8835EB board";
 	compatible = "sprd,sp8835eb";
-	sprd,sc-id = <8830 1 0x20000>;
+	qcom,msm-id = <8830 1 0x20000>;
 	#address-cells = <1>;
 	#size-cells = <1>;
 	interrupt-parent = <&gic>;
diff --git a/arch/arm/boot/dts/sprd-scx20_j1mini3g_r02.dts b/arch/arm/boot/dts/sprd-scx20_j1mini3g_r02.dts
index 8d8ad48a..73c28dba 100644
--- a/arch/arm/boot/dts/sprd-scx20_j1mini3g_r02.dts
+++ b/arch/arm/boot/dts/sprd-scx20_j1mini3g_r02.dts
@@ -33,7 +33,7 @@
 / {
 	model = "Spreadtrum SP8835EB board";
 	compatible = "sprd,sp8835eb";
-	sprd,sc-id = <8830 2 0x20000>;
+	qcom,msm-id = <8830 2 0x20000>;
 	#address-cells = <1>;
 	#size-cells = <1>;
 	interrupt-parent = <&gic>;
diff --git a/arch/arm/boot/dts/sprd-scx20_j1mini3g_r03.dts b/arch/arm/boot/dts/sprd-scx20_j1mini3g_r03.dts
index 9690db02..becf5503 100644
--- a/arch/arm/boot/dts/sprd-scx20_j1mini3g_r03.dts
+++ b/arch/arm/boot/dts/sprd-scx20_j1mini3g_r03.dts
@@ -33,7 +33,7 @@
 / {
 	model = "Spreadtrum SP8835EB board";
 	compatible = "sprd,sp8835eb";
-	sprd,sc-id = <8830 3 0x20000>;
+	qcom,msm-id = <8830 3 0x20000>;
 	#address-cells = <1>;
 	#size-cells = <1>;
 	interrupt-parent = <&gic>;
