--- a/anbox-init.sh
+++ b/anbox-init.sh
@@ -18,12 +18,12 @@ set -x
 function prepare_filesystem() {
 	# These dev files need to be adjusted everytime as they are
 	# bind mounted into the temporary rootfs
-	for f in qemu_pipe qemu_trace goldfish_pipe input/* ; do
+	for f in qemu_pipe qemu_trace goldfish_pipe input input/* ; do
 		if [ ! -e "/dev/$f" ] ; then
 			continue
 		fi
 		chown system:system /dev/$f
-		chmod 0666 /dev/$f
+		chmod 0777 /dev/$f
 	done
 
 	if [ -e "/dev/tun" ] ; then
