diff --git a/drivers/tty/vt/selection.c b/drivers/tty/vt/selection.c
index 8d7ab8809..499139552 100644
--- a/drivers/tty/vt/selection.c
+++ b/drivers/tty/vt/selection.c
@@ -348,8 +348,8 @@ int set_selection(const struct tiocl_selection __user *v, struct tty_struct *tty
 	return ret;
 }
 
-int set_selection(const struct tiocl_selection __user *v,
-	struct tty_struct *tty)
+/*
+int set_selection(const struct tiocl_selection __user *v, struct tty_struct *tty)
 {
 	int ret;
 
@@ -359,6 +359,7 @@ int set_selection(const struct tiocl_selection __user *v,
 
 	return ret;
 }
+*/
 
 /* Insert the contents of the selection buffer into the
  * queue of the tty associated with the current console.
