diff --git a/scripts/Makefile.lib b/scripts/Makefile.lib
index 1e67d6a08..a554f41bf 100644
--- a/scripts/Makefile.lib
+++ b/scripts/Makefile.lib
@@ -330,7 +330,7 @@ $(obj)/%.dtb.S: $(obj)/%.dtb
 quiet_cmd_dtc = DTC     $@
 cmd_dtc = mkdir -p $(dir ${dtc-tmp}) ; \
 	$(CPP) $(dtc_cpp_flags) -x assembler-with-cpp -o $(dtc-tmp) $< ; \
-	$(srctree)/scripts/dtc/dtc_overlay -@ -O dtb -o $@ -b 0 \
+	$(srctree)/out/scripts/dtc/dtc -@ -O dtb -o $@ -b 0 \
 		$(addprefix -i,$(dir $<) $(DTC_INCLUDE)) $(DTC_FLAGS) \
 		-d $(depfile).dtc.tmp $(dtc-tmp) 2>$@.dtout || $(call show_dtc_error, $@.dtout) ;\
 	$(DTC) -q -O dts -I dtb -o $@.reverse.dts $@ ; \
