diff --git a/hildon/hildon-color-chooser.c b/hildon/hildon-color-chooser.c
index 1dd0293..ad7188d 100644
--- a/hildon/hildon-color-chooser.c
+++ b/hildon/hildon-color-chooser.c
@@ -109,19 +109,19 @@ init_borders                                    (GtkWidget *w,
                                                  GtkBorder *inner, 
                                                  GtkBorder *outer);
 
-inline void 
+void 
 inline_clip_to_alloc                            (void *s, 
                                                  GtkAllocation *a);
 
-inline void 
+void 
 inline_sub_times                                (GTimeVal *result, 
                                                  GTimeVal *greater, 
                                                  GTimeVal *lesser);
 
-inline void 
+void 
 inline_limited_expose                           (HildonColorChooser *self);
 
-inline void 
+void 
 inline_draw_hue_bar                             (GtkWidget *widget, 
                                                  int x, 
                                                  int y, 
@@ -130,7 +130,7 @@ inline_draw_hue_bar                             (GtkWidget *widget,
                                                  int sy, 
                                                  int sh);
 
-inline void
+void
 inline_draw_hue_bar_dimmed                      (GtkWidget *widget, 
                                                  int x, 
                                                  int y, 
@@ -139,28 +139,28 @@ inline_draw_hue_bar_dimmed                      (GtkWidget *widget,
                                                  int sy, 
                                                  int sh);
 
-inline void 
+void 
 inline_draw_sv_plane                            (HildonColorChooser *self, 
                                                  int x, 
                                                  int y, 
                                                  int w, 
                                                  int h);
 
-inline void 
+void 
 inline_draw_sv_plane_dimmed                     (HildonColorChooser *self, 
                                                  int x, 
                                                  int y, 
                                                  int w, 
                                                  int h);
 
-inline void 
+void 
 inline_draw_crosshair                           (unsigned char *buf, 
                                                  int x, 
                                                  int y, 
                                                  int w, 
                                                  int h);
 
-inline void 
+void 
 inline_h2rgb                                    (unsigned short hue, 
                                                  unsigned long *rgb);
 
@@ -462,7 +462,7 @@ hildon_color_chooser_unmap                      (GtkWidget *widget)
     GTK_WIDGET_CLASS (parent_class)->unmap (widget);
 }
 
-inline void 
+void 
 inline_clip_to_alloc                            (void *s, 
                                                  GtkAllocation *a)
 {
@@ -574,7 +574,7 @@ hildon_color_chooser_expose                     (GtkWidget *widget,
 }
 
 
-inline void 
+void 
 inline_sub_times                                (GTimeVal *result, 
                                                  GTimeVal *greater, 
                                                  GTimeVal *lesser)
@@ -588,7 +588,7 @@ inline_sub_times                                (GTimeVal *result,
     }
 }
 
-inline void 
+void 
 inline_limited_expose                           (HildonColorChooser *sel)
 {
     GTimeVal curr_time, result;
@@ -849,7 +849,7 @@ hildon_color_chooser_set_color                  (HildonColorChooser *chooser,
     g_signal_emit (chooser, color_chooser_signals[COLOR_CHANGED], 0);
 }
 
-inline void
+void
 inline_h2rgb                                    (unsigned short hue, 
                                                  unsigned long *rgb)
 {
@@ -963,7 +963,7 @@ intern_h2rgb8                                   (unsigned short hue,
 
 /* optimization: do not ask hue for each round but have bilinear vectors */
 /* rethink: benefits from handling data 8 bit? (no shift round) */
-inline void 
+void 
 inline_draw_hue_bar                             (GtkWidget *widget, 
                                                  int x, 
                                                  int y, 
@@ -1024,7 +1024,7 @@ inline_draw_hue_bar                             (GtkWidget *widget,
     g_free(buf);
 }
 
-inline void 
+void 
 inline_draw_hue_bar_dimmed                      (GtkWidget *widget, 
                                                  int x, 
                                                  int y, 
@@ -1075,7 +1075,7 @@ inline_draw_hue_bar_dimmed                      (GtkWidget *widget,
     gdk_draw_pixbuf (widget->parent->window, widget->style->fg_gc [0], priv->dimmed_bar, 0, 0, x, y, w, h, GDK_RGB_DITHER_NONE, 0, 0);
 }
 
-inline void 
+void 
 inline_draw_crosshair                           (unsigned char *buf, 
                                                  int x, 
                                                  int y, 
@@ -1106,7 +1106,7 @@ inline_draw_crosshair                           (unsigned char *buf,
     }
 }
 
-inline void 
+void 
 inline_draw_sv_plane                            (HildonColorChooser *sel, 
                                                  int x, 
                                                  int y, 
@@ -1179,7 +1179,7 @@ inline_draw_sv_plane                            (HildonColorChooser *sel,
     g_free(buf);
 }
 
-inline void 
+void 
 inline_draw_sv_plane_dimmed                     (HildonColorChooser *sel, 
                                                  int x, 
                                                  int y, 
