From c40db6ea8e9a1c9d7d136de44108fbacd659fd45 Mon Sep 17 00:00:00 2001
From: Steven Noonan <steven@uplinklabs.net>
Date: Wed, 17 Nov 2021 11:55:18 -0800
Subject: [PATCH 11/21] x86: touch clocksource watchdog after syncing TSCs

Signed-off-by: Steven Noonan <steven@uplinklabs.net>
[Forward port to 6.5]
Signed-off-by: Cristian Ciocaltea <cristian.ciocaltea@collabora.com>
---
 arch/x86/kernel/smpboot.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/x86/kernel/smpboot.c b/arch/x86/kernel/smpboot.c
index 3f57ce68a3f1..f13ddbbb5d79 100644
--- a/arch/x86/kernel/smpboot.c
+++ b/arch/x86/kernel/smpboot.c
@@ -60,6 +60,7 @@
 #include <linux/stackprotector.h>
 #include <linux/cpuhotplug.h>
 #include <linux/mc146818rtc.h>
+#include <linux/clocksource.h>
 
 #include <asm/acpi.h>
 #include <asm/cacheinfo.h>
@@ -1248,6 +1249,7 @@ void arch_thaw_secondary_cpus_begin(void)
 
 void arch_thaw_secondary_cpus_end(void)
 {
+	clocksource_touch_watchdog();
 	cache_aps_init();
 }
 
@@ -1282,6 +1284,8 @@ void __init native_smp_cpus_done(unsigned int max_cpus)
 {
 	pr_debug("Boot done\n");
 
+	clocksource_touch_watchdog();
+
 	calculate_max_logical_packages();
 	build_sched_topology();
 	nmi_selftest();
-- 
2.44.0

