diff --git a/src/core/config/linux.pri b/src/core/config/linux.pri
index 076d8a3..22a894f 100644
--- a/src/core/config/linux.pri
+++ b/src/core/config/linux.pri
@@ -6,6 +6,13 @@ gn_args += \
     use_cups=false \
     use_gconf=false \
     use_gio=false \
+    use_x11=false \
+    use_ozone=true \
+    ozone_auto_platforms=false \
+    ozone_platform_headless=true \
+    ozone_platform_external=true \
+    ozone_platform=\"qt\" \
+    enable_palmbridge=true \
     use_gnome_keyring=false \
     use_kerberos=false \
     linux_use_bundled_binutils=false
