diff --git a/KTp/CMakeLists.txt b/KTp/CMakeLists.txt
index 85d578b..f7b68f9 100644
--- a/KTp/CMakeLists.txt
+++ b/KTp/CMakeLists.txt
@@ -93,6 +93,8 @@ target_link_libraries (KTpCommonInternals
                 PUBLIC
                     KF5::Wallet #included from wallet-interface.h
                     ${TELEPATHY_QT5_LIBRARIES}
+		    Qt5::DBus
+		    Qt5::Xml
                 PRIVATE
                     ${ktp_private_LIBS}
                     KF5::KIOWidgets
diff --git a/KTp/OTR/CMakeLists.txt b/KTp/OTR/CMakeLists.txt
index 12e28de..b245f6c 100644
--- a/KTp/OTR/CMakeLists.txt
+++ b/KTp/OTR/CMakeLists.txt
@@ -29,6 +29,7 @@ generate_export_header(KTpOTR)
 target_link_libraries (KTpOTR
                         PUBLIC
                             ${TELEPATHY_QT5_LIBRARIES}
+			    Qt5::DBus
                         PRIVATE
                             KF5::I18n
 )
