diff --git a/src/libusermetricsinput/CMakeLists.txt b/src/libusermetricsinput/CMakeLists.txt
index bf3262e..47b7346 100644
--- a/src/libusermetricsinput/CMakeLists.txt
+++ b/src/libusermetricsinput/CMakeLists.txt
@@ -79,6 +79,7 @@ qt5_use_modules(
 target_link_libraries(
 	usermetricsinput
 	usermetricscommon
+	intl
 )
 
 set_target_properties(
@@ -157,5 +158,6 @@ qt5_use_modules(
 target_link_libraries(
 	usermetricsinput-increment
 	usermetricsinput
+	intl
 )
 
