diff --git a/kcmkwin/kwinscripts/main.cpp b/kcmkwin/kwinscripts/main.cpp
index f5ee04b..baa5175 100644
--- a/kcmkwin/kwinscripts/main.cpp
+++ b/kcmkwin/kwinscripts/main.cpp
@@ -22,3 +22,5 @@
 
 K_PLUGIN_FACTORY(KcmKWinScriptsFactory,
                  registerPlugin<Module>("kwin-scripts");)
+
+#include "main.moc"
diff --git a/kcmkwin/kwinscripts/module.cpp b/kcmkwin/kwinscripts/module.cpp
index a0d698e..ccf7d41 100644
--- a/kcmkwin/kwinscripts/module.cpp
+++ b/kcmkwin/kwinscripts/module.cpp
@@ -40,8 +40,6 @@
 
 #include "version.h"
 
-K_PLUGIN_FACTORY_DECLARATION(KcmKWinScriptsFactory)
-
 Module::Module(QWidget *parent, const QVariantList &args) :
     KCModule(parent, args),
     ui(new Ui::Module),
