diff --git a/src/3rdparty/chromium/third_party/yasm/BUILD.gn b/qtwebengine-opensource-src-5.9.1-patched/src/3rdparty/chromium/third_party/yasm/BUILD.gn
index 6f49a02..0703793 100644
--- a/src/3rdparty/chromium/third_party/yasm/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/yasm/BUILD.gn
@@ -317,7 +317,7 @@ if (current_toolchain == host_toolchain) {
         "-pedantic",
       ]
     }
-
+    libs = [ "intl" ]
     # TODO(ajwong): This should take most of the generated output as
     # inputs.
     deps = [
