diff --git a/libdbustest/Makefile.am b/libdbustest/Makefile.am
index d76ff52..d048a83 100644
--- a/libdbustest/Makefile.am
+++ b/libdbustest/Makefile.am
@@ -40,7 +40,7 @@ libdbustest_la_CFLAGS = \
 	-DBUSTLE_DUAL_MONITOR="\"$(pkgdatadir)/dbus-test-bustle-handler\"" \
 	-DWATCHDOG="\"$(pkglibexecdir)/dbus-test-watchdog\"" \
 	-DG_LOG_DOMAIN=\"libdbustest\" \
-	-Wall -Werror -Wextra
+	-Wall -Wextra
 
 libdbustest_la_LIBADD = \
 	libdbustest-generated.la \
@@ -59,7 +59,7 @@ libdbustest_generated_la_CFLAGS = \
 	$(DBUS_TEST_RUNNER_CFLAGS) \
 	-I$(builddir) \
 	-DG_LOG_DOMAIN=\"libdbustest\" \
-	-Wall -Werror
+	-Wall
 
 libdbustest_generated_la_SOURCES = \
 	dbus-mock-iface.h \
diff --git a/src/Makefile.am b/src/Makefile.am
index 24a3f1f..7c44544 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -6,7 +6,7 @@ dbus_test_runner_CFLAGS  = $(DBUS_TEST_RUNNER_CFLAGS) \
 	$(COVERAGE_CFLAGS) \
 	-I$(top_srcdir) \
 	-DDEFAULT_SESSION_CONF="\"$(datadir)/dbus-test-runner/session.conf\"" \
-	-Wall -Werror -Wextra
+	-Wall -Wextra
 dbus_test_runner_LDADD   = $(DBUS_TEST_RUNNER_LIBS) \
 	$(top_builddir)/libdbustest/libdbustest.la
 dbus_test_runner_LDFLAGS = $(COVERAGE_LDFLAGS)
diff --git a/tests/Makefile.am b/tests/Makefile.am
index 7eccc63..4f4aad9 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -245,7 +245,7 @@ test_own_name_SOURCES = \
 	test-own-name.c
 test_own_name_CFLAGS = \
 	$(DBUS_TEST_RUNNER_CFLAGS) \
-	-Wall -Werror
+	-Wall
 test_own_name_LDADD = \
 	$(DBUS_TEST_RUNNER_LIBS)
 
@@ -253,7 +253,7 @@ test_check_name_SOURCES = \
 	test-check-name.c
 test_check_name_CFLAGS = \
 	$(DBUS_TEST_RUNNER_CFLAGS) \
-	-Wall -Werror
+	-Wall
 test_check_name_LDADD = \
 	$(DBUS_TEST_RUNNER_LIBS)
 
@@ -351,7 +351,7 @@ test_libdbustest_CFLAGS = \
 	-I$(top_srcdir) \
 	-DSESSION_CONF="\"$(top_srcdir)/data/session.conf\"" \
 	-DGETNAME_PATH="\"$(abs_builddir)/test-libdbustest-getname\"" \
-	-Wall -Werror
+	-Wall
 test_libdbustest_LDADD = \
 	$(DBUS_TEST_RUNNER_LIBS) \
 	$(top_builddir)/libdbustest/libdbustest.la
@@ -360,7 +360,7 @@ test_libdbustest_getname_SOURCES = \
 	test-libdbustest-getname.c
 test_libdbustest_getname_CFLAGS = \
 	$(DBUS_TEST_RUNNER_CFLAGS) \
-	-Wall -Werror
+	-Wall
 test_libdbustest_getname_LDADD = \
 	$(DBUS_TEST_RUNNER_LIBS)
 
@@ -389,7 +389,7 @@ test_libdbustest_mock_CFLAGS = \
 	-I$(top_srcdir) \
 	-DSESSION_CONF="\"$(top_srcdir)/data/session.conf\"" \
 	-DGETNAME_PATH="\"$(abs_builddir)/test-libdbustest-getname\"" \
-	-Wall -Werror
+	-Wall
 test_libdbustest_mock_LDADD = \
 	$(DBUS_TEST_RUNNER_LIBS) \
 	$(top_builddir)/libdbustest/libdbustest.la
