From 764885a2d850600d1e7ca50a54271ba2d42374d0 Mon Sep 17 00:00:00 2001
From: Pablo Castellano <pablo@anche.no>
Date: Mon, 17 Jul 2017 20:04:24 +0200
Subject: [PATCH 3/3] compositor-fbdev: print the pixman type guessed in
 calculate_pixman_format()

---
 libweston/compositor-fbdev.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libweston/compositor-fbdev.c b/libweston/compositor-fbdev.c
index 8158841..4ee3279 100644
--- a/libweston/compositor-fbdev.c
+++ b/libweston/compositor-fbdev.c
@@ -241,6 +241,8 @@ calculate_pixman_format(struct fb_var_screeninfo *vinfo,
 	         vinfo->green.offset >= vinfo->red.offset)
 		type = PIXMAN_TYPE_ABGR;
 
+	weston_log("Type guessed: %i:\n", type);
+
 	if (type == PIXMAN_TYPE_OTHER)
 		return 0;
 
-- 
2.7.4

