diff --git a/synchronizer/main.cpp b/synchronizer/main.cpp
index f1709bc..5595929 100644
--- a/synchronizer/main.cpp
+++ b/synchronizer/main.cpp
@@ -55,6 +55,7 @@ static Listener *listener = nullptr;
 void printStacktrace()
 {
 #ifndef Q_OS_WIN
+#if defined(__GNU_LIBRARY__)
     int skip = 1;
 	void *callstack[128];
 	const int nMaxFrames = sizeof(callstack) / sizeof(callstack[0]);
@@ -90,6 +91,7 @@ void printStacktrace()
     }
     std::cerr << trace_buf.str();
 #endif
+#endif
 }
 
 static int sCounter = 0;
