diff --git a/meson.build b/meson.build
index 80e1fc70e..1a3ca564f 100644
--- a/meson.build
+++ b/meson.build
@@ -1,10 +1,10 @@
 project('mutter', 'c',
-  version: '45.rc',
+  version: '45_rc',
   meson_version: '>= 0.60.0',
   license: 'GPLv2+'
 )
 
-split_version = meson.project_version().split('.')
+split_version = meson.project_version().split('_')
 
 # API version, bump each development cycle
 libmutter_api_version = '13'
