--- a/mipi-dbi-cmd
+++ b/mipi-dbi-cmd
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 
 # SPDX-License-Identifier: CC0-1.0
@@ -12,7 +12,6 @@
 # You should have received a copy of the CC0 Public Domain Dedication along with this software.
 # If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 
-from __future__ import print_function
 import argparse
 import sys
 
