From be5e9f092e96a0cdd759e517133146a28f10c093 Mon Sep 17 00:00:00 2001
From: Clayton Craft <clayton@craftyguy.net>
Date: Mon, 31 Jan 2022 16:37:13 -0800
Subject: [PATCH] librem5: enable console on UART4

---
 include/configs/librem5.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/librem5.h b/include/configs/librem5.h
index d826f856e1..46b07ba5be 100644
--- a/include/configs/librem5.h
+++ b/include/configs/librem5.h
@@ -59,7 +59,7 @@
 
 #define CONFIG_USB_GADGET_VBUS_DRAW	2
 
-#define CONSOLE_ON_UART1
+#define CONSOLE_ON_UART4
 #define CONFIG_BAUDRATE			115200
 
 #ifdef CONSOLE_ON_UART1
-- 
2.35.1

