diff --git a/src/solid/devices/backends/udisks2/udisksblock.h b/src/solid/devices/backends/udisks2/udisksblock.h
--- a/src/solid/devices/backends/udisks2/udisksblock.h
+++ b/src/solid/devices/backends/udisks2/udisksblock.h
@@ -24,9 +24,7 @@
 #include <solid/devices/ifaces/block.h>
 #include "udisksdeviceinterface.h"
 
-#ifdef Q_OS_FREEBSD
-#include <sys/types.h>
-#endif
+#include <sys/types.h> // dev_t
 
 namespace Solid
 {

