By ollieparanoid
---
diff --git a/src/ioslaves/file/sharefd_p.h b/src/ioslaves/file/sharefd_p.h
index 664f507..4e5f8ec 100644
--- a/src/ioslaves/file/sharefd_p.h
+++ b/src/ioslaves/file/sharefd_p.h
@@ -75,8 +75,13 @@ public:
         : io_buf{0}
         , cmsg_buf{0}
         , io{io_buf, sizeof io_buf}
-        , msg{nullptr, 0,  &io, 1,  &cmsg_buf, sizeof cmsg_buf, 0}
     {
+        // musl libc has a padding in msghdr, so initialize by name
+        memset(&msg, 0, sizeof(msghdr));
+        msg.msg_iov = &io;
+        msg.msg_iovlen = 1;
+        msg.msg_control = &cmsg_buf;
+        msg.msg_controllen = sizeof cmsg_buf;
     }

     msghdr *message()
