diff -Naur a/drivers/gpu/msm/kgsl_iommu.c b/drivers/gpu/msm/kgsl_iommu.c
--- a/drivers/gpu/msm/kgsl_iommu.c	2013-11-29 03:09:34.000000000 +0100
+++ b/drivers/gpu/msm/kgsl_iommu.c	2019-05-26 00:59:09.026481139 +0200
@@ -520,7 +520,7 @@
 	return status;
 }
 
-inline unsigned int kgsl_iommu_sync_lock(struct kgsl_mmu *mmu,
+static inline unsigned int kgsl_iommu_sync_lock(struct kgsl_mmu *mmu,
 						unsigned int *cmds)
 {
 	struct kgsl_device *device = mmu->device;
@@ -571,7 +571,7 @@
 	return cmds - start;
 }
 
-inline unsigned int kgsl_iommu_sync_unlock(struct kgsl_mmu *mmu,
+static inline unsigned int kgsl_iommu_sync_unlock(struct kgsl_mmu *mmu,
 					unsigned int *cmds)
 {
 	struct kgsl_device *device = mmu->device;
