From c87e8bea9e31c921de5d9707f471c3eee479cb0a Mon Sep 17 00:00:00 2001
From: Robert Yang <decatf@gmail.com>
Date: Wed, 10 Oct 2018 17:52:32 -0400
Subject: [PATCH 2/2] Make systemd optional

---
 configure.ac | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 689946f..97e9f7a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -32,7 +32,12 @@ AC_ARG_WITH([udevrulesdir],
 	    [with_udevrulesdir=$($PKG_CONFIG --variable=udevdir udev)"/rules.d"])
 AC_SUBST([udevrulesdir], [$with_udevrulesdir])
 
-PKG_CHECK_EXISTS(systemd, [], [AC_MSG_ERROR(systemd development libraries are required)])
+AC_ARG_ENABLE([systemd],
+	      [AS_HELP_STRING([--disable-systemd],[do not build systemd service])],
+	      [],[enable_systemd=yes])
+if test x$enable_systemd = xyes; then
+	PKG_CHECK_EXISTS(systemd, [], [AC_MSG_ERROR(systemd development libraries are required)])
+fi
 AC_ARG_WITH([systemdsystemunitdir],
 	    AS_HELP_STRING([--with-systemdsystemunitdir=DIR], [Directory for systemd service files]),
 	    [],
-- 
2.17.1

