--- a/drivers/input/touchscreen/mediatek/FT5446/focaltech_driver.c
+++ b/drivers/input/touchscreen/mediatek/FT5446/focaltech_driver.c
@@ -598,12 +598,12 @@ static void tpd_down(int x, int y, int p) {
 	}
 	input_report_key(tpd->dev, BTN_TOUCH, 1);
 	input_report_abs(tpd->dev, ABS_MT_TOUCH_MAJOR, 20);
-	input_report_abs(tpd->dev, ABS_MT_PRESSURE, 0x3f);
+	//input_report_abs(tpd->dev, ABS_MT_PRESSURE, 0x3f);
 	input_report_abs(tpd->dev, ABS_MT_POSITION_X, x);
 	input_report_abs(tpd->dev, ABS_MT_POSITION_Y, y);
 	//printk("tpd:D[%4d %4d %4d] ", x, y, p);
 	/* track id Start 0 */
-	//input_report_abs(tpd->dev, ABS_MT_TRACKING_ID, p); 
+	input_report_abs(tpd->dev, ABS_MT_TRACKING_ID, p); 
 	input_mt_sync(tpd->dev);
 	if (FACTORY_BOOT == get_boot_mode()|| RECOVERY_BOOT == get_boot_mode())
 	{   
