From 1091e7af92eba4bba90632e9ea3386ae68dbeba2 Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@z3ntu.xyz>
Date: Wed, 3 Oct 2018 17:08:08 +0200
Subject: [PATCH 3/5] Add option to disable building of docs

---
 ubuntu-sdk.pro | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/ubuntu-sdk.pro b/ubuntu-sdk.pro
index 4cd2e9cfe..b5ff7e0db 100644
--- a/ubuntu-sdk.pro
+++ b/ubuntu-sdk.pro
@@ -11,7 +11,10 @@ load(qt_parts)
 src_uitk_launcher.subdir = ubuntu-ui-toolkit-launcher
 src_uitk_launcher.depends = sub-src
 
-SUBDIRS += po documentation app-launch-profiler src_uitk_launcher apicheck
+SUBDIRS += po app-launch-profiler src_uitk_launcher apicheck
+!CONFIG(no_docs) {
+    SUBDIRS += documentation
+}
 
 sub_tests.CONFIG -= no_default_target
 sub_tests.CONFIG -= no_default_install
-- 
2.20.1

